File: Warehousing.Data/Entities/Inventory.cs
namespace Warehousing.Data.Entities
{
    public class Inventory : BaseClass
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public Product? Product { get; set; }

        public int StoreId { get; set; }
        public Store? Store { get; set; }
        
        // Variant support
        public ProductVariant? Variant { get; set; }
        public int? VariantId { get; set; }

        public decimal Quantity { get; set; }   // total stock in that store
    }
}

File: Warehousing.UI/src/app/admin/models/Inventory.ts
import { Product } from "./product";
import { Store } from "./store";

export interface Inventory {
    id: number;
    productId: number;
    product: Product | null;
    storeId: number;
    store: Store | null;
    variantId?: number | null; // Added to support variant-specific inventory
    quantity: number;
    createdAt: string | null;
    createdBy: string | null;
    updatedAt: string | null;
    updatedBy: string | null;
}